graphium/nn/architectures/global_architectures.py

```diff
@@ -1324,6 +1324,12 @@ def _recursive_divide_dim(x: collections.abc.Mapping):
                     _recursive_divide_dim(v)
                 elif k in ["in_dim", "out_dim", "in_dim_edges", "out_dim_edges"]:
                     x[k] = round(v / divide_factor)
+                elif k in ["embed_dim"]:
+                    num_heads = x.get("num_heads", 1)
+                    x[k] = round(v / divide_factor)
+                    assert (
+                        x[k] % num_heads == 0
+                    ), f"embed_dim={x[k]} is not divisible by num_heads={num_heads}"
```

Comment:
I don't think it's needed since there's another assertion in `AttentionLayerMup`. @maciej-sypetkowski can you check if we remove that part when scaling by a factor that's not divisible by `num_heads`, does it still work?

Reply:
Yes, it's not needed